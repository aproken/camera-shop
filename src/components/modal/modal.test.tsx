import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Modal from './modal';

describe('Тестирование компонента Modal', () => {
  it('Открытие модального окна', () => {
    const mockHide = jest.fn(() => null);
    const {container} = render(<Modal isOpen hide={mockHide}>test</Modal>);

    expect(container).toMatchSnapshot();
  });

  it('Закрытие', async () => {
    const mockHide = jest.fn(() => null);
    const {container} = render(<Modal isOpen hide={mockHide}>test</Modal>);

    // Закрытие на крестик
    fireEvent.click(screen.getByRole('button'));
    // Закрытие на overlay
    fireEvent.click(screen.getByLabelText('Закрыть попап', {selector: '.modal__overlay'}));
    // Закрытие на Esc
    fireEvent.keyDown(container, {key: 'Escape'});

    await waitFor(() => {
      expect(mockHide.mock.calls).toHaveLength(3);
    });
  });
});

