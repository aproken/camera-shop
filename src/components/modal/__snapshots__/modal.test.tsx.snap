// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Modal Открытие модального окна 1`] = `<div />`;
